File: C++/minNumDisappeared.cpp
#include<iostream>
#include<vector>

class Solution{
    public:
    int minNumberDisappeared(std::vector<int>& nums) {
        int n =nums.size();
        for(int i=0;i<n;i++){
            if(nums[i]<=0||nums[i]>n){
                nums[i] =n+1;
            }
        }
        for(int i=0;i<n;i++){
            int num =abs(nums[i]);
            if(num>n)continue;
            int idx =num-1;
            if(nums[idx]>0){
                nums[idx]=-nums[idx];
            }
        }
        for(int i=0;i<n;i++){
            if(nums[i]>0){
                return i+1;
            }
        }
        return n+1;
    }
};

int main(){
    std::vector<int>nums={-2,3,4,1,5};
    Solution solution;
    std::cout<<solution.minNumberDisappeared(nums)<<std::endl;
    return 0;
}

File: python/minNumDisappeared.py
class Solution:
    def minNumberDisappeared(self , nums: list[int]) -> int:
        n = len(nums)
        
        for i in range(n):
            if nums[i] <= 0 or nums[i] > n:
                nums[i] = n + 1

        for i in range(n):
            num = abs(nums[i])
            if num > n:
                continue
            idx = num - 1
            if nums[idx] > 0:
                nums[idx] = -nums[idx]

        for i in range(n):
            if nums[i] > 0:
                return i + 1

        return n + 1

nums=[-2,3,4,1,5]
s=Solution()
print(s.minNumberDisappeared(nums))
